docs/DeveloperGuide.md

```diff
@@ -2,37 +2,254 @@
+
+#### Illness
+
+This class is inherited by all other illnesses. It contains the name of the illness `illnessName`, its `severityLevel`,
+whether it is chronic `isChronic`, and most importantly, the list of symptoms specific to the illness `symptoms`.
+
+Child classes of `Illness` are given their values on creation of the object in their constructor function.
+Hence `new Diarrhoea()` will create an object with `.getSymptoms()` returning an `ArrayList<Symptom>`
+of `Symptom.WET_STOOL`, `Symptom.STOMACH_ACHE`, and `Symptom.DIARRHOEA`.
+
+Knowing this, each child object of `Illness` will be created and stored in `ArrayList<Illness> ALL_ILLNESSES`.
+
+#### Diagnosis algorithm
+
+![DiagnosisSequenceDiagram.png](diagrams%2FDiagnosisSequenceDiagram.png)
+
+After the patient has entered all their symptoms, it will be passed to the `getPossibleIllnesses()` function
+in `Diagnosis` where it will match each symptom the patient has to each symptom of all possible
+illnesses `ALL_ILLNESSES`.
+
+The percentage of matches the patient's symptoms has to the total number of symptoms the illness is specified to exhibit
+is calculated, and if this percentage does not cross the `POSSIBILITY_THRESHOLD`, the patient would be declared to
+possibly have the illness. To store this likely candidate, an `IllnessMatch` object is created storing the
+candidate `Illness` and the `similarityPercentage`.
+
+An `ArrayList<IllnessMatch> possibleIllnesses` is then returned to retrieve the appropriate medicine in another class.
+
+### Implementation of MedicineManager class, Medicine class
+
+![img_1.png](diagrams%2FMedicineSequenceDiagram.png)
+
+*Sequence diagram for medicine class*
+
+#### From the menu to the Medicine Manager
+
+User input from the menu will first get parsed by the parser class.
+From the Parser class, a new MedicineManager class is created. The MedicineManager class initialises 2 hash tables
+with medicines created with the Medicine class. Each medicine has a different name and dosage, which can be
+retrieved using the Medicine.toString() and Medicine.getDosages().
+
+#### Medicines
+
+Medicine information is stored in two hash tables. One for matching the relevant medication with illnesses, the other
+matching the medicines with correct dosages.
+
+#### Analysing the symptoms and prescribing the medications
+
+Diagnosis.getIllnessMatch() is called in the MedicineManager class, which returns an ArrayList of illnessMatch
+objects. IllnessMatch objects contain information about the illness' name and the likelihood that the symptoms
+the user has inputted match the illness as explained in the "Diagnosis" section above.
+
+Each illness will be analysed in the analyseIllness() method, and the medications that cure that illness
+will be suggested to the user. This information is then printed, and stored in the user's account.
 
+[comment]: <> (//@@Thunderdragon221)
+### Storage
+
+![StorageClassDiagram.png](diagrams%2FStorageClassDiagram.png)
```

Comment:
Should the inclusion of exceptions in class diagrams be made consistent across all class diagrams?